### PR TITLE
Taran golnoush/fragment library book ui

### PR DESCRIPTION
**Pull request by ghassanzadeh**
Added the .xml file for the CardView

Screenshot_20201021_151857

### CODE REVIEW

**Comment by ghassanzadeh** [comment]
i need to modify the id parts since we don't use findviewbyid().

File: app/src/main/res/drawable/status_color.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval"
+    >
+    <solid
+        android:color="#4CAF50"/>
```
Comment (calebschoepp) [inline app/src/main/res/drawable/status_color.xml:5]: Pretty sure there is a `colors.xml` file that `#4CAF50` can be extracted to as a resource e.g. name it something like `green`

File: app/src/main/res/layout/fragment_library_book.xml [hunk 2]
```diff
@@ -5,38 +5,78 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="10dp"
+    android:layout_marginBottom="10dp"
+    android:layout_marginLeft="10dp"
+    android:layout_marginRight="10dp"
+    android:checkable="true"
+    android:clickable="true"
+    android:focusable="true"
+    app:cardElevation="5dp">
```
Comment (calebschoepp) [inline app/src/main/res/layout/fragment_library_book.xml:18]: IMO we should extract this as a standard style resource in `styles.xml` and use it for every card in the app.

Thoughts @cyrusdiego ?

File: app/src/main/res/layout/fragment_library_book.xml [hunk 3]
```diff
@@ -5,38 +5,78 @@
...
+                android:id="@+id/bookTitle"
+                android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_margin="@dimen/text_margin"
-                android:textAppearance="?attr/textAppearanceListItem"
-                android:text="@{book.title}" />
+                android:layout_marginTop="15dp"
+                android:padding="5dp"
+                android:text="Title"
+                android:textAppearance="?android:textAppearanceMedium"
```
Comment (calebschoepp) [inline app/src/main/res/layout/fragment_library_book.xml:51]: Just curious, what does the `?` indicate?

File: app/src/main/res/layout/fragment_library_book.xml [hunk 4]
```diff
@@ -5,38 +5,78 @@
...
+                android:textColor="@android:color/black"
+                android:textSize="25dp" />
 
             <TextView
-                android:id="@+id/content"
-                android:layout_width="wrap_content"
+                android:id="@+id/bookAuthor"
```
Comment (calebschoepp) [inline app/src/main/res/layout/fragment_library_book.xml:56]: Are we standardizing on camelCase or snake_case for ids @cyrusdiego @armiantos ?

**Review by calebschoepp — CHANGES_REQUESTED** [review]
Great work guys. Left a couple of comments

File: app/src/main/res/layout/fragment_library_book.xml (hunk 2 above)
Comment (cyrusdiego) [reply app/src/main/res/layout/fragment_library_book.xml:18]: Yeah, I think extracting these properties to a common style is a good idea, we would most likely have 3 types of "cards" (library, unlibrary, and exchange) and they may all have this common style.

File: app/src/main/res/layout/fragment_library_book.xml (hunk 4 above)
Comment (cyrusdiego) [reply app/src/main/res/layout/fragment_library_book.xml:56]: I think snake_case. By default, android studios use snake_case for ids, strings, etc.

File: app/src/main/res/layout/fragment_library_book.xml [hunk 5]
```diff
@@ -5,38 +5,78 @@
...
+                android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_margin="@dimen/text_margin"
-                android:textAppearance="?attr/textAppearanceListItem"
-                android:text="@{book.title}" />
+                android:layout_marginTop="15dp"
+                android:padding="5dp"
+                android:text="Title"
+                android:textAppearance="?android:textAppearanceMedium"
+                android:textColor="@android:color/black"
```
Comment (cyrusdiego) [inline app/src/main/res/layout/fragment_library_book.xml:52]: I think these can be extracted to the style.xml as well or material UI may have default values we can just use with their theming?

**Review by cyrusdiego — CHANGES_REQUESTED** [review]
Good job guys, left some comments as well.

File: app/src/main/res/layout/fragment_library_book.xml (hunk 3 above)
Comment (ghassanzadeh) [reply app/src/main/res/layout/fragment_library_book.xml:51]: that's just the syntax to use textAppearance. We can use textsize too.

**Comment by ghassanzadeh** [comment]
Thanks @cyrusdiego and @calebschoepp, will modify the file with these feedbacks.

File: app/src/main/res/layout/fragment_library_book.xml (hunk 2 above)
Comment (armiantos) [reply app/src/main/res/layout/fragment_library_book.xml:18]: Either way, the margins could've been simplified to `android:layout_margin` I believe.

File: app/src/main/res/layout/fragment_library_book.xml (hunk 3 above)
Comment (armiantos) [reply app/src/main/res/layout/fragment_library_book.xml:51]: ? indicates you want to use style, aka `@android:style` I believe

File: app/src/main/res/layout/fragment_library_book.xml [hunk 6]
```diff
@@ -5,38 +5,78 @@
...
-            android:layout_width="wrap_content"
+        android:layout_height="150dp"
+        android:orientation="horizontal"
+        android:paddingTop="10dp"
+        android:paddingBottom="10dp">
+
+        <RelativeLayout
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="2">
```
Comment (armiantos) [inline app/src/main/res/layout/fragment_library_book.xml:30]:
```suggestion
            android:layout_weight="2"
            android:layout_gravity="center">
```
to center the picture in the layout

File: app/src/main/res/layout/fragment_library_book.xml [hunk 7]
```diff
@@ -5,38 +5,78 @@
...
-                android:layout_width="wrap_content"
+                android:id="@+id/bookTitle"
+                android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_margin="@dimen/text_margin"
-                android:textAppearance="?attr/textAppearanceListItem"
-                android:text="@{book.title}" />
+                android:layout_marginTop="15dp"
+                android:padding="5dp"
+                android:text="Title"
```
Comment (armiantos) [inline app/src/main/res/layout/fragment_library_book.xml:50]:
```suggestion
                android:text="@{book.title}"
```
To get everyone familiar with data binding

**Review by armiantos — CHANGES_REQUESTED** [review]
Other than the small nitpicks, it looks good guys!

File: app/src/main/res/layout/fragment_library_book.xml (hunk 7 above)
Comment (armiantos) [reply app/src/main/res/layout/fragment_library_book.xml:50]: You can use `@{book.author}` too now

**Comment by armiantos** [comment]
Closes #31

**Comment by ghassanzadeh** [comment]
close #42

**Comment by ghassanzadeh** [comment]
Closes #42

**Pull request closed without merging**